app/services/processors/external/contributions.rb

```diff
@@ -0,0 +1,21 @@
+module Processors
+  module External
+    class Contributions < BaseService
+      def call
+        usernames.each do |username|
+          repositories_data = Processors::External::Repositories.call(username)
+          repositories_data.each do |repository_data|
+            project = Builders::ExternalProject.call(repository_data)
+            Processors::External::PullRequests.call(project, username)
```

Comment:
im not sure if i understood, we could change the check of arrays in response by status to know what we should return, is that what you mean?

Reply:
Yeah, I'm sorry, I'll clarify on that.
A few weeks ago I added the `:raise_error` middleware to the requests made to Github, so if the response is not successful, an error will be raised.
We need to consider that any failed request will break the code execution, so we need to plan ahead and think of what we should do in such cases.
What I have opted to do is to let all processes related to that single request fail (because I could do nothing with the failed response) and catch the error when other processes that could finish correctly may be compromised. And track the error occurrence in Exception Hunter.
You can check an example [here](https://github.com/rootstrap/rs-code-review-metrics/blob/develop/app/services/processors/open_source_project_views_updater.rb), where I catch the error and let the process using that class continue with the execution normally.